Improve Discovery in provisioning (#2890)
theforeman/foreman-documentation: guides/common/modules/proc_performing-discovery-in-pxeless-mode.adoc

[id="performing-discovery-in-pxeless-mode_{context}"]
= Performing Discovery in PXE-less mode
[[Implementing_PXE_less_Discovery_{context}]]

> Discovery in PXE-less mode uses the Discovery ISO and requires you to attend to the process.

Reviewer: ```suggestion
Discovery in PXE-less mode uses the Discovery ISO and requires you to pay attention to the process.
```

Author: What's wrong with "attend"? When it's attended, they don't only pay attention, they have to perform some steps manually. How about "requires your manual intervention" then?

For example, to copy to a USB drive at `/dev/sdb`:
+
[options="nowrap" subs="+quotes"]
----
# dd bs=4M \
if=/usr/share/foreman-discovery-image/foreman-discovery-image-_version_.iso \
of=/dev/sdb
----
. Insert the Discovery boot media into a host, start the host, and boot from the media.
. The Discovery image displays options for either *Manual network setup* or *Discovery with DHCP*:

* *Manual network setup*:
include::snip_step-discovery-pxeless-primary-interface.adoc[]
.. On the *Network configuration* screen, enter the *Address*, *Gateway*, and *DNS*.
Hit *Next* to continue.
* *Discovery with DHCP*:
include::snip_step-discovery-pxeless-primary-interface.adoc[]
.. The Discovery image attempts to automatically configure the network interface by using a DHCP server, such as one that a {SmartProxyServer} provides.
. On the *Credentials* screen, enter the following options:
+
** In the *Server URL* field, enter the URL of {ProjectServer} or Discovery {SmartProxyServer}.
If you refer to a {SmartProxyServer}, include the {SmartProxy} port number.
** In the *Connection type* field, select the connection type: `Server` for {ProjectServer} or `Foreman Proxy` for {SmartProxyServer}.

+

> Hit *Next* to continue.

Reviewer: ```suggestion
Click *Next* to continue.
```

Author: This is a problem. The GUI, on which this is done, doesn't have the mouse cursor. The user has to select the button by using the keyboard and then press Enter.

> . Optional: On the *Custom facts* screen, enter custom facts for the Facter tool to relay back to {ProjectServer}.

Reviewer: Is it called a `Facter tool`?

Author: TBH, I have no clue what this is :))

Enter a name and value for each custom fact you need.

> . Hit *Confirm* to proceed.

Reviewer: ```suggestion
. Click *Confirm* to proceed.
```

[id="performing-discovery-in-pxeless-mode_{context}"]
= Performing Discovery in PXE-less mode
[[Implementing_PXE_less_Discovery_{context}]]

Discovery in PXE-less mode uses the Discovery ISO and requires you to attend to the process.

.Prerequisites
ifdef::satellite[]
* You have installed the Foreman Discovery image.
For more information, see xref:Installing_the_Discovery_Service_{context}[].
endif::[]
ifndef::satellite[]
* You have installed the Foreman Discovery image or built a custom Discovery image ISO.
For more information, see one of the following resources:
** xref:Installing_the_Discovery_Service_{context}[]
** xref:Building_a_Discovery_Image_{context}[]
endif::[]

.Procedure

> . Copy the Discovery ISO to either a CD, DVD, or a USB stick.

Reviewer 2: ```suggestion
. Copy the Discovery ISO to a CD, DVD, or a USB flash drive.
```
"or" is only used if there are 2 options.

For example, to copy to a USB stick at `/dev/sdb`:
+
[options="nowrap" subs="+quotes"]
----
# dd bs=4M \
if=/usr/share/foreman-discovery-image/foreman-discovery-image-_version_.iso \
of=/dev/sdb
----
. Insert the Discovery boot media into a bare-metal host, start the host, and boot from the media.
The Discovery Image displays an option for either *Manual network setup* or *Discovery with DHCP*:
+

> * If you select *Manual network setup*, the Discovery image requests a set of network options.

Reviewer 2: ```suggestion
* *Manual network setup*: Enter the following options:

** Primary network interface
** IPv4 Address
** etc.
```

Reviewer 2: It would be a good idea to find out what is on the screen. 

Otherwise, we're basically telling the user, "If you select 'x', fill in the options. If you select 'y', fill in different options." :-) 

If the options themselves are not important, you could change the lead-in sentence to a step: "Select **Manual network setup** or **Discovery with DHCP**, enter the required fields, and click **Next**."

This includes the primary network interface that connects to {ProjectServer}.
This Discovery image also asks for network interface configuration options, such as an *IPv4 Address*, *IPv4 Gateway*, and an *IPv4 DNS* server.
+
After entering these details, select *Next*.

> * If you select *Discovery with DHCP*, the Discovery image requests only the primary network interface that connects to {ProjectServer}.

Reviewer 2: ```suggestion
**Discovery with DHCP*: Enter the primary network interface that connects to {ProjectServer}.
```

It attempts to automatically configure the network interface using a DHCP server, such as one that a {SmartProxyServer} provides.

> . After the primary interface configuration, the Discovery image requests the *Server URL*, which is the URL of {ProjectServer} or {SmartProxyServer} offering the Discovery service.

Reviewer 2: ```suggestion
. Enter the URL and port of {ProjectServer} or {SmartProxyServer} offering the Discovery service.
```
No need for "After the configuration" or "the Discovery image requests". This is a procedure. Also, I'm not sure you need an example. But that's your call.

For example, to use the integrated {SmartProxy} on {ProjectServer}, use the following URL:
+
[options="nowrap" subs="+quotes,attributes"]
----
https://{foreman-example-com}:{smartproxy_port}
----

> . Set the *Connection type* to `Proxy`, then select *Next*.

Reviewer 2: ```suggestion
. Enter `Proxy` as the connection type and then click *Next*.
```
UI field names are used as adjectives, not nouns. If "Next" is a button, the correct verb is "click".

> . Optional: The Discovery image also provides a set of fields to input *Custom facts* for the Facter tool to relay back to {ProjectServer}.

Reviewer 2: ```suggestion
. Optional: Enter custom facts in *name*-*value* format for the Facter tool and click *Confirm*.
```
